### ISSUE TITLE
app is frozen 

### ISSUE THREAD

**Issue by erickv121** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#467]
When I first launched it it was working but when I changed one or two of the settings (Enable microphone on application startup and Start as Maximized ) it stop working I have relaunch it and even uninstalled it. Three time and nothing is working at all there is nothing popping up. Nothing i change in the settings work anymore it doesn't even save them i can change the theme and just nothing is happening

**Comment by coder-lg** [comment]
Can you show a screenshot?

**Comment by erickv121** [comment]
<img width="1270" alt="Screenshot 2021-07-29 024223" src="https://user-images.githubusercontent.com/36559421/127444303-1ce40242-ba45-4052-8a64-db48b0cfbf68.png">

**Comment by erickv121** [comment]
Nothing i type in the search works. Can't change anything in settings

**Comment by coder-lg** [comment]
@erickv121 have you tried building the program via NPM? If not, try it.

**Comment by erickv121** [comment]
No i have not what is that?

**Comment by coder-lg** [comment]
@erickv121 wait, i am making a gist in GH, you download it and run it. It is a .bat file that automates your entire process and builds the app for you.

**Comment by coder-lg** [comment]
<script src="https://gist.github.com/coder-lg/f82b7337ac76ed6d70c2bd8e8dd7600d.js"></script>

**Comment by sumeshir26** [comment]
@coder-lg Wow this is nice, now whenever someone asks about building this project, can I share this gist?

**Comment by coder-lg** [comment]
@erickv121 https://gist.github.com/coder-lg/f82b7337ac76ed6d70c2bd8e8dd7600d is my gist, you can download and run it :)
study it if you want too!

**Comment by coder-lg** [comment]
> @coder-lg Wow this is nice, now whenever someone asks about building this project, can I share this gist?

https://gist.github.com/coder-lg/f82b7337ac76ed6d70c2bd8e8dd7600d there it is! One way to find out if this is helpful too!

**Comment by coder-lg** [comment]
Main caveat: You need Git. I will try making one for Mac and Linux too, gotta study `sh`

**Comment by coder-lg** [comment]
> @coder-lg Wow this is nice, now whenever someone asks about building this project, can I share this gist?

yup

**Comment by coder-lg** [comment]
Since, a batch file in Windows just automates your normal CMD process, I will try methods of `curl`ing the file in CMD and check if you really need to download it too.

**Comment by sumeshir26** [comment]
@coder-lg Is git on winget? You might be able to use it

**Comment by sumeshir26** [comment]
Or can you not use winget with a bat file? I am not at all an expert in these things

**Comment by coder-lg** [comment]
> @coder-lg Is git on winget? You might be able to use it

No, it is a standalone app, as far as I know. But, the batch file I made uses only Git and the Windows CMD.

**Comment by erickv121** [comment]
hey what do i do now with it?

**Comment by coder-lg** [comment]
Just, open notepad, paste the source code, get Git, rename it to `run.bat`, store in your desktop, and run it.

**Comment by erickv121** [comment]
yeah yeah but what do i do then?

**Comment by coder-lg** [comment]
Go to Desktop, go to Google Assistant folder you cloned, then go to the dist directory, and open `win-unpacked`. Then, run Google Assistant.exe

**Comment by coder-lg** [comment]
First of all, @erickv121 did the batch file work successfully?

**Comment by erickv121** [comment]
No i don't think it did 
<img width="673" alt="Screenshot 2021-07-29 043802" src="https://user-images.githubusercontent.com/36559421/127460251-78b8d68b-77e1-4588-929f-a524e6b41bca.png">

**Comment by coder-lg** [comment]
ooookay. Lemme edit that file. and @sumeshir26 can you test [`this`](https://gist.github.com/coder-lg/7751ed92c59e31792c31345dc63618a6) because I don't have a Linux machine.

**Comment by coder-lg** [comment]
> No i don't think it did
> <img alt="Screenshot 2021-07-29 043802" width="673" src="https://user-images.githubusercontent.com/36559421/127460251-78b8d68b-77e1-4588-929f-a524e6b41bca.png">

@erickv121 brb

**Comment by coder-lg** [comment]
@erickv121 try [`this`](https://gist.github.com/coder-lg/f82b7337ac76ed6d70c2bd8e8dd7600d)

**Comment by coder-lg** [comment]
usual method like last time

**Comment by coder-lg** [comment]
Atleast something was working last time. Share the screenshot when the next error comes, I will fix it.

**Comment by coder-lg** [comment]
> When I first launched it it was working but when I changed one or two of the settings (Enable microphone on application startup and Start as Maximized ) it stop working I have relaunch it and even uninstalled it. Three time and nothing is working at all there is nothing popping up. Nothing i change in the settings work anymore it doesn't even save them i can change the theme and just nothing is happening

If it still fails after building the app, try deleting `tokens.json`

**Comment by erickv121** [comment]
I just did that and I'm still not getting any response from the app

**Comment by coder-lg** [comment]
Did you follow this code?
```bat
@ECHO OFF
echo Get Git. && echo If you didn't, the batch file will fail. && echo Move this to your user directory. && cd Desktop && git clone https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client.git && cd Google-Assistant-Unofficial-Desktop-Client && npm install && npm run dist && echo built! && echo open this directory in Explorer, && echo go to `dist` and then go to `win-unpacked`. && echo Then, run Google Assistant.exe! && echo Voila!
PAUSE
```
I tested it and now it works. You have a user profile directory, right? Put the file there and try it.

**Comment by coder-lg** [comment]
> I just did that and I'm still not getting any response from the app

Wait, @Melvin-Abraham got some solution related to config. Let me find it or @Melvin-Abraham might give a solution.

**Comment by coder-lg** [comment]
Aha! Found it! Now, go to `C:\Users\{your-username}\AppData\Roaming\Google Assistant` where {your-username} is your Windows Username. Delete each file inside the folder and delete the folder itself! Now, try reinstalling and sign in. I guess this solves the problem.

**Comment by erickv121** [comment]
................ 
 
It works :/

**Comment by coder-lg** [comment]
You meddled here with your settings, that was what caused the problem

**Comment by erickv121** [comment]
LMAO i didn't think "meddled" with Start as Maximized would froze my app lol

**Comment by erickv121** [comment]
but hey whatever thanks for the help

**Comment by coder-lg** [comment]
hey @erickv121 or @Melvin-Abraham can anybody close this issue and mark it as solved? because, it got solved like somewhat yesterday, so can you close this issue?

**Comment by coder-lg** [comment]
> LMAO i didn't think "meddled" with Start as Maximized would froze my app lol

oh yeah meddled, by meaning, ***accidentally*** corrupting the `config.json` file.

**Comment by behnammhmd** [comment]
> 
> 
> When I first launched it it was working but when I changed one or two of the settings (Enable microphone on application startup and Start as Maximized ) it stop working I have relaunch it and even uninstalled it. Three time and nothing is working at all there is nothing popping up. Nothing i change in the settings work anymore it doesn't even save them i can change the theme and just nothing is happening

can confirm. The same happened to me.

**Comment by Melvin-Abraham** [comment]
> hey @erickv121 or @Melvin-Abraham can anybody close this issue and mark it as solved? because, it got solved like somewhat yesterday, so can you close this issue?

The problems that occurred could be due to some internal issues which I think must be fixed before closing this issue as resolved.

**Comment by Melvin-Abraham** [comment]
@erickv121 If you have any error logs in the DevTools (<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>I</kbd>) with those settings enabled please do share it here so that the root cause could be addressed...

**Comment by im-coder-lg** [comment]
Okay... So is there any fix because this does happen sometimes if you **accidentally** corrupt the config.json. You told about deleting the config folder and trying again, so that's what I told here.

**Comment by Melvin-Abraham** [comment]
> So is there any fix because this does happen sometimes if you accidentally corrupt the config.json

In order to understand and fix the issue, some sort of logs would be required to find the root cause initially.

> You told about deleting the config folder and trying again, so that's what I told here

This is actually a new bug report and thus unfortunately has not been implemented in the codebase. So, manually setting the config.json would do the job for now but after an update, the issue would still come up if it's not addressed from the root. :)

**Comment by im-coder-lg** [comment]
:thinking: what if you add a special button in the GAUDC Settings that delete the config? Just like the token part, why not do that? Even add another function that shows the GAUDC Config path, like for Windows, a specific component, for macOS and Linux, a specific component. Is that possible?

**Comment by Melvin-Abraham** [comment]
@im-coder-lg That's why there's **Fallback Mode** available which could be accessed from either the settings or tray menu. But its use is only recommended in some specific cases.

**Comment by im-coder-lg** [comment]
But, how about a path displayer?
